src/Measurements.jl

```diff
@@ -1,108 +1,47 @@
-struct Measurement{DataT, ObsT <: MonteCarloObservable.AbstractObservable}
-    obs::ObsT
-    data::DataT
-    init!::Function
-    update_data!::Function
-    measure!::Function
-    finish!::Function
```

Comment:
Note that `Function` is an abstract type and should be avoided as a field type.

Reply:
The code generation in `init!` actually dereferences all the individual `update_data!` and `measure!` calls. Meaning that if you set the `measure!` field to some function `f`, the generated `measure!` will contain `f`, not `measurement.measure!`.

And that is also why the code generation was there in the first place. Iirc, parameterizing on `Function` did not help.